components/brave_shields/browser/tracking_protection_service.cc

```diff
@@ -124,10 +320,10 @@ TrackingProtectionService::GetThirdPartyHosts(const std::string& base_host) {
     }
   }
 
-  char* thirdPartyHosts =
-    tracking_protection_client_->findFirstPartyHosts(base_host.c_str());
+  std::string thirdPartyHosts =
+      tracking_protection_client_->findFirstPartyHosts(base_host.c_str());
```

Comment:
this is returning `char*` that was allocated on heap... it gets leaked, because it's never cleaned up. Delete should still be called

A work-around for this (if you wanted to use `std::string`) could be:
```
char* thirdPartyHosts =
    tracking_protection_client_->findFirstPartyHosts(base_host.c_str());
std::string thirdPartyHostsStr = std::string(thirdPartyHosts);
delete []thirdPartyHosts;
// use thirdPartyHostsStr now instead
```
```